app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,150 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class HomeViewModel(
+    private val movieRepository: MovieRepository,
+    private val homeScreenMapper: HomeScreenMapper,
+) : ViewModel() {
+
+    private val popularCategoryLabels = listOf(
+        MovieCategory.POPULAR_STREAMING,
+        MovieCategory.POPULAR_ONTV,
+        MovieCategory.POPULAR_FORRENT,
+        MovieCategory.POPULAR_INTHEATHERS
+    )
+
+    private val nowPlayingCategoryLabels = listOf(
+        MovieCategory.NOWPLAYING_MOVIES,
+        MovieCategory.NOWPLAYING_TV
+    )
+
+    private val upcomingCategoryLabels = listOf(
+        MovieCategory.UPCOMING_TODAY,
+        MovieCategory.UPCOMING_THISWEEK
+    )
+
+
+    private val popularMoviesCategorySelected = MutableStateFlow(HomeMovieCategoryViewState(listOf(), listOf()))
+    val popularViewState: StateFlow<HomeMovieCategoryViewState> = popularMoviesCategorySelected.asStateFlow()
+
+    private val nowPlayingMoviesCategorySelected = MutableStateFlow(HomeMovieCategoryViewState(listOf(), listOf()))
+    val nowPlayingViewState: StateFlow<HomeMovieCategoryViewState> = nowPlayingMoviesCategorySelected.asStateFlow()
+
+    private val upcomingMoviesCategorySelected = MutableStateFlow(HomeMovieCategoryViewState(listOf(), listOf()))
+    val upcomingViewState: StateFlow<HomeMovieCategoryViewState> = upcomingMoviesCategorySelected.asStateFlow()
+
+    init {
+        getPopularMovies()
+        getNowPlayingMovies()
+        getUpcomingMovies()
+    }
+
+    private fun getPopularMovies() {
+        viewModelScope.launch {
+            movieRepository.popularMovies(MovieCategory.POPULAR_STREAMING).collect {
+                popularMoviesCategorySelected.emit(
+                    homeScreenMapper.toHomeMovieCategoryViewState(
+                        movieCategories = popularCategoryLabels,
+                        selectedMovieCategory = MovieCategory.POPULAR_STREAMING,
+                        movies = it
+                    )
+                )
+            }
+        }
+    }
```

Comment:
Instead of these `get_x_Movies()` functions in init, you can just map from movie repository to `_x_viewState`, the same way you did in `MovieDetailsViewModel` and `FavoritesViewModel`. Just add `_x_MoviesCategorySelected` flow before it. Then you'll have something like this:

```
val popularCategoryViewState = popularMoviesCategorySelected
            .flatMapLatest { selectedMovieCategory ->
                movieRepository.popularMovies(selectedMovieCategory)
                    .map { ... }
            }
```